lang/syn/src/codegen/accounts/constraints.rs

```diff
@@ -1028,6 +1031,25 @@ impl<'a> OptionalCheckScope<'a> {
     }
 }
 
+fn generate_get_token_account_space(mint: &Expr) -> proc_macro2::TokenStream {
```

Comment:
it's probably not a huge deal but everything in here ends up getting called twice because the `generate_crate_account` uses the `space` quote in two places.

Reply:
Also not sure if it makes a difference but do we want the `anchor_spl` to be imported using `::anchor_spl` just to make sure it comes from the right place. (I could be wrong here)

Reply:
> it's probably not a huge deal but everything in here ends up getting called twice because the generate_crate_account uses the space quote in two places.

Ah nice spot, I've fixed this up to only run once. Since the token-2022 case involves a deserialization of the mint, among other things, it's much better to just do it once.

> Also not sure if it makes a difference but do we want the anchor_spl to be imported using ::anchor_spl just to make sure it comes from the right place. (I could be wrong here)

Also a great call, done

Reply:
Done in [3b63905](https://github.com/coral-xyz/anchor/pull/2386/commits/3b639057c2265527e17f5908a4e4855478d19464)